BrownianMotion/Continuity/Chaining.lean

```diff
@@ -175,20 +175,85 @@ lemma edist_chainingSequence_pow_two_le {ε₀ : ℝ≥0∞} (hC : ∀ i, IsCove
   gcongr <;> simpa only [inv_eq_one_div] using ENNReal.sum_geometric_two_le _
 
 lemma scale_change {F : Type*} [PseudoEMetricSpace F] (hC : ∀ i, IsCover (C i) (ε i) A)
-    (m : ℕ) (hm : m ≤ k) (X : E → F) (δ : ℝ≥0∞) :
+    (m : ℕ) (X : E → F) (δ : ℝ≥0∞) :
     ⨆ (s) (t) (_hs : s ∈ C k) (_ht : t ∈ C k) (_h : edist s t ≤ δ), edist (X s) (X t)
-    ≤ ⨆ (s) (t) (hs : s ∈ C k) (ht : t ∈ C k) (_h : edist s t ≤ δ),
-        edist (X (chainingSequence hC hs m)) (X (chainingSequence hC ht m))
+    ≤ (⨆ (s) (t) (hs : s ∈ C k) (ht : t ∈ C k) (_h : edist s t ≤ δ),
+        edist (X (chainingSequence hC hs m)) (X (chainingSequence hC ht m)))
       + 2 * ⨆ (s) (hs : s ∈ C k), edist (X s) (X (chainingSequence hC hs m)) := by
-  sorry
+  -- We will be using `iSup_add` later, so we need to bundle the suprema so that we only take
+  -- suprema over nonempty index types.
+  conv_lhs => congr; ext; rw [iSup_comm]
+  conv_lhs => rw [iSup_subtype']; congr; ext; rw [iSup_subtype', iSup_subtype']
+  conv_rhs => congr; congr; ext; rw [iSup_comm]
+  conv_rhs => congr; rw [iSup_subtype']; congr; ext; rw [iSup_subtype', iSup_subtype']
+  conv_rhs => right; right; rw [iSup_subtype']
```

Comment:
Can you extract that rewriting of the supremum as a lemma? It could be often useful if we keep the statements as they are not.

Reply:
Done.

Reply:
Thanks!